torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -430,11 +453,20 @@ class ScriptModuleSerializer {
     for (const auto& td : data_pickle.tensorData()) {
       WriteableTensorData writable_td = getWriteableTensorData(td);
       std::string fname = prefix + c10::to_string(i++);
-      writer_.writeRecord(fname, writable_td.data(), writable_td.sizeInBytes());
+      if (supportTensorsArchiveTable) {
+        const auto found = tensors_archive_table_.find(td);
+        if (found == tensors_archive_table_.end()) {
+          writer_.writeRecord(
+              fname, writable_td.data(), writable_td.sizeInBytes());
```

Comment:
So this effectively writes only non tensor constants?

Reply:
That's correct. Only tensor constants for now. The tensor size is the main reason for the size increase. I thought about more general case (like for all `IValue`), but `IValue` includes too many different types and the hash function will be different. We can make the hash function more general if the model size is still an issue. I feel like we might not gain much though.

Reply:
I agree. I was just asking a clarifying question. By the way if you run the model through optimize_for_mobile or freeze_module you might see bigger savings since all parameters tensors will be converted to constant as well.